nav2_costmap_2d/include/nav2_costmap_2d/collision_checker.hpp

```diff
@@ -61,7 +60,7 @@ class CollisionChecker
 
   // Name used for logging
   std::string name_;
-  nav2_util::GetRobotPoseClient & get_robot_pose_client_;
+  std::shared_ptr<tf2_ros::Buffer> tf_;
```

Comment:
maybe no need to keep a pointer, in this case seems like a reference is good enough